language-design.md

```diff
@@ -17,7 +17,78 @@ For each quote, describe what it was about the quote that led you pick it.
+supports a particularly preactical feature that Java does not, made me have a lot more respect for Python.
+I had always thought of Python as more of a language for learners because (I hear) it has so much overhead
+(which is maybe still significant), but its ease of use coupled with the ability to overload its unique,
+natural language-feeling syntax is pretty awesome/powerful. My first thoughts went to the `in` operator,
+but in all honesty, Java has done a great job of making iteration through a group of objects feel simpler.
+In Java 5, the 'enhanced for loop' was added, which makes Java iteration feel very similar to iteration in Python
+--though it's still missing a sort of natural language feeling.
+
+<!--
+* More respect for python because it supports operator overloading
+* Operator overloading is possible in C++/Python but not in Java
+* Operator overloading lets an extension of a language feel more familiar, convenient, and 'primitive:' like the language being extended.
+* I never really considered the importance of operator overloading before, but I think it makes sense given the described benefits that users should be able to warp operators and keywords for personal use.
+-->
+
+> ...users will not wait for 'The Right Thing,' but will use what comes first and put up with the warts. *But*--users will not put up with the warts for all time. It is not long till they scream and moan and beg for changes. The small language will grow the warts will be shaved off or patched. [Steele, 1998]
+
+We agree with Steele's idea that DSLs, or "small languages," (or really _any_ language) should be created,
+at least to some extent, with flexibility in mind: languages tend to evolve, not only because the language designers decide to
+update language features or philosophies, but also because the values and needs of the language's users change and adapt.
+In the same vein, the adoption and continued use of a language can be perceived as a sort of function of the ability for users
+to affect the direction and development of the language. Domain-specific needs change along with technology and society,
+and the language will need to evolve and keep pace with the change,
+or else risk fading to the background and being replaced by another language that--at least for the moment--
+meets users' needs and expectations. When we came into this class, we had our own ideals about programming languages and DSLs,
+in particular that a language if based on a particularly fruitful and foundational philosophy could be 'the right tool' for users;
+but as with most things, designers and developers must continue to communicate with their users
+and gauge the reception of their product in order to continue tuning the language to meet user expectations
+whilst fixing the language within some philosophy or other boundary. This reading has been enlightening about
+_how_ one should go about proposing, creating, and maintaining a DSL.
```

Comment:
> In the same vein, the adoption and continued use of a language can be perceived as a sort of function of the ability for users to affect the direction and development of the language.

Do you feel languages like C++ do this adequately? If not, what's the balance between "need to grow to survive" and "legacy wins"?